cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisioningService.java

```diff
@@ -421,7 +403,13 @@ private Runnable createProvisionTask(ProvisioningTaskInfo taskInfo, Provisioner
     ProvisionerContext context;
     try {
       context = createContext(programRunId, taskInfo.getUser(), taskInfo.getProvisionerProperties(),
-                              createSSHContext(taskInfo.getSecureKeyInfo()));
+                              new DefaultSSHContext(locationFactory.create(taskInfo.getSecureKeysDir()),
+                                                    createSSHKeyPair(taskInfo)));
+    } catch (IOException e) {
+      runWithProgramLogging(taskInfo.getProgramRunId(), taskInfo.getProgramOptions().getArguments().asMap(),
+                            () -> LOG.error("Failed to load ssh key. The run will be marked as failed.", e));
+      provisionerNotifier.deprovisioned(taskInfo.getProgramRunId());
```

Comment:
this should actually call deprovisioning instead of deprovisioned, since the code can be called when resuming a task on CDAP restart. The existing call for invalid macro is wrong, can you change that as well?

Reply:
I see. I copied that from invalid macro :)

Reply:
yep, figured that was the case.